src/Expression/SubExpression.purs

```diff
@@ -20,7 +19,9 @@ removeSubExpressions = removeSubExpressionsWithMap []
     ExpressionVariable name -> case VM.lookup variableMap name of
       Just value -> value
       _ -> ExpressionVariable name
-    ExpressionLet name expression parentExpression -> removeSubExpressionsWithMap (variableMap <> [ (Tuple name expression) ]) parentExpression
+    ExpressionLet name expression parentExpression -> removeSubExpressionsWithMap (variableMap <> [ (Tuple name (removeSubExpressionsWithMap variableMap expression)) ]) parentExpression
```

Comment:
I think the following naming will be more consistent with the rest and avoid the potentially misleading concept of "parent":
```suggestion
    ExpressionLet name nameExpression expression -> removeSubExpressionsWithMap (variableMap <> [ (Tuple name (removeSubExpressionsWithMap variableMap nameExpression)) ]) expression
```

Reply:
Resolved in latest commit